addons/sale_project/models/project.py

```diff
@@ -340,20 +341,49 @@ def _get_profitability_sale_order_items_domain(self, domain=None):
     def _get_revenues_items_from_sol(self, domain=None, with_action=True):
         sale_line_read_group = self.env['sale.order.line'].sudo()._read_group(
             self._get_profitability_sale_order_items_domain(domain),
-            ['product_id', 'ids:array_agg(id)', 'untaxed_amount_to_invoice', 'untaxed_amount_invoiced'],
-            ['product_id'],
+            ['product_id', 'is_downpayment', 'ids:array_agg(id)', 'untaxed_amount_to_invoice', 'untaxed_amount_invoiced'],
+            ['product_id', 'is_downpayment'], lazy=False
         )
         display_sol_action = with_action and len(self) == 1 and self.user_has_groups('sales_team.group_sale_salesman')
         revenues_dict = {}
         total_to_invoice = total_invoiced = 0.0
+        data = []
+        sequence_per_invoice_type = self._get_profitability_sequence_per_invoice_type()
         if sale_line_read_group:
-            sols_per_product = {
-                res['product_id'][0]: (
-                    res['untaxed_amount_to_invoice'],
-                    res['untaxed_amount_invoiced'],
-                    res['ids'],
-                ) for res in sale_line_read_group
-            }
+            sols_per_product = {}
+            downpayment_amount_invoiced = 0
+            downpayment_sol_ids = []
+            for res in sale_line_read_group:
+                if res['is_downpayment']:
+                    downpayment_amount_invoiced += res['untaxed_amount_invoiced']
+                    downpayment_sol_ids += res['ids']
+                else:
+                    sols_per_product[res['product_id'][0]] = (
+                            res['untaxed_amount_to_invoice'],
+                            res['untaxed_amount_invoiced'],
+                            res['ids'],
+                        )
+            if downpayment_amount_invoiced:
+                downpayments_data = {
+                    'id': 'downpayments',
+                    'sequence': sequence_per_invoice_type['downpayments'],
+                    'invoiced': downpayment_amount_invoiced,
+                    'to_invoice': -downpayment_amount_invoiced
+                }
+                if with_action and self.user_has_groups('sales_team.group_sale_salesman_all_leads'):
```

Comment:
Why don't we need to check if there are some ids? :thinking:

Reply:
There's already an if statement above to ensure there is an amount for the 'downpayments' section. I can change the 'if downpayment_amount_invoiced' to 'if downpayment_sol_ids' if you think its neater that way.

Reply:
Ho ok I see. :astonished:
Hm, maybe, yes. Because, in case (now or in the future) we can have downpayments of 0, checking the id rather than the amount sounds more solid. What do you think?

Reply:
well, for now at least, it is not possible. Because the downpayments SOL are locked and can not be modified. And if it get changed, i think its better to do the check with the amount, because i dont see the utility of adding a section in the project updates, with an amount to invoice of 0. Might as well just not show the section. There are already too many sections in that panel (imo)

Reply:
You're right :ok_hand: